encodings/utf16.js

```diff
@@ -112,6 +112,11 @@ function Utf16Decoder(options, codec) {
 
 Utf16Decoder.prototype.write = function(buf) {
     if (!this.decoder) {
+        // Support Uint8Array
+        if (!Buffer.isBuffer(buf)) {
+            buf = Buffer.from(buf)
+        }
```

Comment:
Ultimately I'd want to get rid of `Buffer.concat` so that we can avoid conversion to Buffers at all. Meanwhile, I think this stop gap is fine. (Same with utf32).